Makefile

```diff
@@ -78,14 +73,40 @@ endif
 # Binaries.
 #
 # Note: Need to use abspath so we can invoke these from subdirectories
-KUSTOMIZE := $(abspath $(TOOLS_BIN_DIR)/kustomize)
-SETUP_ENVTEST := $(abspath $(TOOLS_BIN_DIR)/setup-envtest)
-CONTROLLER_GEN := $(abspath $(TOOLS_BIN_DIR)/controller-gen)
-GOTESTSUM := $(abspath $(TOOLS_BIN_DIR)/gotestsum)
-GOLANGCI_LINT := $(abspath $(TOOLS_BIN_DIR)/golangci-lint)
-CONVERSION_GEN := $(abspath $(TOOLS_BIN_DIR)/conversion-gen)
+KUSTOMIZE_VER := v3.9.1
+KUSTOMIZE_BIN := kustomize
+KUSTOMIZE := $(abspath $(TOOLS_BIN_DIR)/$(KUSTOMIZE_BIN)-$(KUSTOMIZE_VER))
+
+SETUP_ENVTEST_VER := v0.0.0-20211110210527-619e6b92dab9
+SETUP_ENVTEST_BIN := setup-envtest
+SETUP_ENVTEST := $(abspath $(TOOLS_BIN_DIR)/$(SETUP_ENVTEST_BIN)-$(SETUP_ENVTEST_VER))
+
+CONTROLLER_GEN_VER := v0.7.1-0.20211110210727-ab52f76cc7d1
+CONTROLLER_GEN_BIN := controller-gen
+CONTROLLER_GEN := $(abspath $(TOOLS_BIN_DIR)/$(CONTROLLER_GEN_BIN)-$(CONTROLLER_GEN_VER))
```

Comment:
How are we going to update to commits in the future? Could we specify a branch here instead?

Reply:
If we only specify a branch here the build wouldn't be reproducible anymore

Reply:
If it's only about how we can calculate the pseudo version we want to use here when we upgrade we could document how to do that, e.g.
```bash
 go list -f '{{.Version}}' -m sigs.k8s.io/controller-tools/cmd/controller-gen@master
```

Reply:
It might be worth having a script to update these whenever necessary rather than leave it to users to do it manually